src/main/java/no/nav/veilarboppfolging/service/AuthService.java

```diff
@@ -235,7 +235,11 @@ public String getInnloggetBrukerToken() {
 
     // NAV ident, fnr eller annen ID
     public String getInnloggetBrukerIdent() {
-        return authContextHolder.getSubject().orElseThrow(() -> new ResponseStatusException(HttpStatus.UNAUTHORIZED, "NAV ident is missing"));
+        return authContextHolder.getIdTokenClaims()
+                .flatMap(claimsSet -> ofNullable((String) claimsSet.getClaim("pid")))
+                .filter(ignored -> authContextHolder.erEksternBruker())
+                .or(authContextHolder::getSubject)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.UNAUTHORIZED, "ident is missing"));
```

Comment:
Burde vi hatt noen enkle tester som sjekker logikken her?

Reply:
Ja, bør testes. Med mindre denne bør lukkes til fordel for å utvide `authContextHolder.getSubject()` til å ta høyde for "pid"-claim i common.

Reply:
Oppdaterte til å heller bruke ny funksjonalitet fra common